projects/igniteui-angular/src/lib/progressbar/circularbar.component.spec.ts

```diff
@@ -11,449 +10,360 @@ const CIRCULAR_OUTER_CLASS = 'igx-circular-bar__outer';
-        expect(bar.step).toBe(expectedValue);
+        it('Value should not exceed the lower limit (0) when operating with floating numbers', () => {
+            progress.value = 0;
+            progress.max = 2.5;
+            progress.value = -0.3;
 
-        const maxVal = 15345;
-        bar.max = maxVal;
-        fix.detectChanges();
+            const expectedRes = 0;
+            expect(progress.value).toBe(expectedRes);
+            expect(progress.valueInPercent).toBe(expectedRes);
 
-        expectedValue = maxVal * ONE_PERCENT;
-        expect(bar.step).toBe(expectedValue);
-    });
+            progress.value = -2;
+            expect(progress.value).toBe(expectedRes);
+            expect(progress.valueInPercent).toBe(expectedRes);
+        });
 
-    it('Value should not exceed the lower limit (0) when operating with floating numbers', () => {
-        const fix = TestBed.createComponent(CircularBarComponent);
-        const compInstance = fix.componentInstance;
-        const bar = compInstance.progressbar;
-        fix.detectChanges();
+        it('Value should not exceed the max limit when operating with floating numbers', () => {
+            let value = 2.67;
+            const max = 2.5;
+            progress.max = max;
+            progress.value = value;
 
-        compInstance.value = 0;
-        fix.detectChanges();
+            expect(progress.value).toBe(max);
+            expect(progress.valueInPercent).toBe(100);
 
-        compInstance.max = 2.5;
-        compInstance.value = -0.3;
-        fix.detectChanges();
+            value = 3.01;
+            progress.value = value;
 
-        const expectedRes = 0;
-        expect(bar.value).toBe(expectedRes);
-        expect(bar.valueInPercent).toBe(expectedRes);
+            expect(progress.value).toBe(max);
+            expect(progress.valueInPercent).toBe(100);
+        });
 
-        compInstance.value = -2;
-        fix.detectChanges();
+        it('when update step is bigger than passed value the progress indicator should follow the value representation', () => {
+            const step = 5;
+            const value = 2;
+            const max = 10;
+            progress.step = step;
+            progress.max = max;
+            progress.value = value;
+
+            const percentValue = toPercent(value, max);
+            expect(progress.value).toBe(value);
+            expect(progress.step).toBe(step);
+            expect(progress.max).toBe(max);
+            expect(progress.valueInPercent).toBe(percentValue);
+
+            // Should not set a step larger than max value
+            progress.step = 20;
+            expect(progress.step).toBe(step);
+            expect(progress.max).toBe(max);
+        });
 
-        expect(bar.value).toBe(expectedRes);
-        expect(bar.valueInPercent).toBe(expectedRes);
-    });
+        it(`when step value is not divisble to passed value the result returned from the value getter should be the same as the passed one`, () => {
```

Comment:
I see that we test increment/decrement functionality which does not include the step property.